src/views/Home.vue

```diff
@@ -0,0 +1,366 @@
+
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          resp.data.response.docs.forEach((product: any) => {
+            this.products[product.productId] = product
+          });
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong"))
+      }
+    },
+
+    getProduct(productId: string) {
+      return this.products[productId] ? this.products[productId] : {}
+    },
+
+    async saveShippingAddressChanges(shipGroup: any) {
+      let resp
+      const payload = {
+        "orderId": this.order.id,
+        "shipGroupSeqId": shipGroup.shipGroupSeqId,
+        "contactMechId": shipGroup.shipTo.postalAddress.id,
+        ...(shipGroup.selectedShipmentMethodTypeId === shipGroup.shipmentMethodTypeId) && { "isEdited": true },
+        "shipmentMethod": `${this.deliveryMethod}@_NA_`,
+        "contactMechPurposeTypeId": "SHIPPING_LOCATION",
+        "facilityId": "WH",
+        "toName": `${shipGroup.editedShipmentAddress.firstName} ${shipGroup.editedShipmentAddress.lastName}`,
+        "address1": shipGroup.editedShipmentAddress.address1,
+        "city": shipGroup.editedShipmentAddress.city,
+        "stateCode": shipGroup.editedShipmentAddress.stateCode,
+        "postalCode": shipGroup.editedShipmentAddress.postalCode,
+        "country": shipGroup.editedShipmentAddress.country
+      }
+
+      try {
+        resp = await OrderService.updateShippingAddress(payload);
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          this.areChangesSaved = true;
+          showToast(translate("Changes saved"))
+        } else {
+          showToast(translate("Something went wrong"))
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong"))
+      }
+    },
+
+    async saveFacilityChanges(shipGroup: any) {
+      let resp
+      const payload = {
+        "orderId": this.order.id,
+        "shipGroupSeqId": shipGroup.shipGroupSeqId,
+        "contactMechId": shipGroup.shipTo.postalAddress.id,
+        "shipmentMethod": `STOREPICKUP@_NA_@CARRIER`,
+        "contactMechPurposeTypeId": "SHIPPING_LOCATION",
+        "facilityId": shipGroup.selectedFacilityId,
+      }
+
+      try {
+        resp = await OrderService.updateFacility(payload);
+        console.log(resp);
+        if (resp.status === 200 && !hasError(resp)) {
+          this.areChangesSaved = true;
+          showToast(translate("Changes saved"))
+        } else {
+          showToast(translate("Something went wrong"))
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong"))
+      }
+    },
+
+    async saveShipmentChanges(shipGroup: any) {
+      shipGroup.selectedShipmentMethodTypeId === 'STOREPICKUP' ? this.saveFacilityChanges(shipGroup) : this.saveShippingAddressChanges(shipGroup);   
+    },
+
+    updateDeliveryMethod(event: any, shipGroup: string) {
```

Comment:
```suggestion
    updateDeliveryMethod(event: any, shipGroup: any) {
```

Remove the explicit casting of shipGroup on line 265